spec/syntax.md

```diff
@@ -152,73 +133,73 @@ which the runtime can use to construct a document tree structure for a UI framew
 
 A message with a single selector:
 
-    {$count :number}
-        1 [You have one notification.]
-        * [You have {$count} notifications.]
+    match {$count :number}
+    when 1 [You have one notification.]
```

Comment:
I really don't care for the use of `when` proposed here. I don't see what value it adds. All of the selectors are clearly part of the `match` statement. The `when` is just extra noise which actually obscures the values.

Note that I still favor delimiting the selectors *and* the literal. The selector delimiter functions the same way that `when` does--identifying that there is an set of values in the `match`.

The single-line example (below, line 143) could then be like:

> match {:platform} [windows] {Settings} [*] {Preferences}

Or (without delims):

> match {:platform} windows {Settings} * {Preferences}

Reply:
I see two benefits to `when`:

1. It allows us to add more keywords in the future that come after the last variant, e.g. `meta` or `attribute`. Usually, programming languages solve this by putting the body of the statement inside a curly-brace-delimited block, but I think we said we didn't want to risk people forgetting the closing brace.

2. (This one is more subjective) It creates a visual correspondence between the `match` and each variant, in particular in case of multiline messages.

Reply:
> Note that I still favor delimiting the selectors and the literal. The selector delimiter functions the same way that when does--identifying that there is an set of values in the match.

Ack here as well. I'm not opposed to delimiting variant keys, but it's true that I'm still hoping that we can use square brackets `[...]` for delimiting patterns (#255).

Technically, I think just relying on order here is good enough: `match <selector list> <variant list> <pattern value> <variant list> <pattern value>` is OK from the parser's point of view, unless we anticipate some other productions to follow all variants in the future (see point 1 in my comment just above).